### DISCUSSION TITLE
Migration Issue: MeteorJS App Fails to Start After Upgrading to Chakra-UI v3

### DISCUSSION THREAD

**Discussion by danyhiol** [chakra-ui/chakra-ui#9280] [Q&A] [answered]
I've been using `Chakra-UI` **v2** without any issues for some time. Recently, I migrated to **Chakra-UI v3**. The migration went smoothly, and everything seemed in place. However, I've encountered a strange issue with my `MeteorJS` app (**version 3**): the app won't start at all, and the problem appears to be related to `@ark-ui/react`. I've tried everything I could think of, including installing `@ark-ui/react` and using Vite, but nothing has worked. At this point, I'm running out of options, and we need to move to production very soon.

I've found [this](https://stackoverflow.com/questions/79172312/chakra-ui-v3-problems-to-run-tests-cannot-find-module-ark-ui-react-format) similar issue on Stackoverflow. 

Any help will be very appreciated.

```
Unable to resolve some modules:

  "@ark-ui/react/format" in                   
.../node_modules/@chakra-ui/react/dist/cjs/components/format/index.cjs
(web.browser.legacy)
  "@ark-ui/react/accordion" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/anatomy" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/dialog" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/editable" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/field" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/file-upload" in              
.../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs (web.browser.legacy)
  "@ark-ui/react/menu" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/popover" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/radio-group" in              
.../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs (web.browser.legacy)
  "@ark-ui/react/rating-group" in             
.../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs (web.browser.legacy)
  "@ark-ui/react/select" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/slider" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/switch" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/avatar" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/checkbox" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/collapsible" in              
.../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs (web.browser.legacy)
  "@ark-ui/react/fieldset" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/hover-card" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/number-input" in             
.../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs (web.browser.legacy)
  "@ark-ui/react/pin-input" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/progress" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/segment-group" in            
.../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs (web.browser.legacy)
  "@ark-ui/react/tooltip" in .../node_modules/@chakra-ui/react/dist/cjs/anatomy.cjs
(web.browser.legacy)
  "@ark-ui/react/color-picker" in             
.../node_modules/@chakra-ui/react/dist/cjs/theme/recipes/color-picker.cjs
(web.browser.legacy)
  "@ark-ui/react/collection" in .../node_modules/@chakra-ui/react/dist/cjs/index.cjs
(web.browser.legacy)
  "@ark-ui/react/clipboard" in                
.../node_modules/@chakra-ui/react/dist/cjs/components/clipboard/clipboard.cjs
(web.browser.legacy)
  "@ark-ui/react/factory" in                  
.../node_modules/@chakra-ui/react/dist/cjs/components/editable/editable.cjs
(web.browser.legacy)
  "@ark-ui/react/environment" in .../node_modules/@chakra-ui/react/dist/cjs/index.cjs
(web.browser.legacy)
  "@ark-ui/react/highlight" in                
.../node_modules/@chakra-ui/react/dist/cjs/components/highlight/highlight.cjs
(web.browser.legacy)
  "@ark-ui/react/locale" in .../node_modules/@chakra-ui/react/dist/cjs/index.cjs
(web.browser.legacy)
  "@ark-ui/react/pagination" in               
.../node_modules/@chakra-ui/react/dist/cjs/components/pagination/pagination.cjs
(web.browser.legacy)
  "@ark-ui/react/portal" in .../node_modules/@chakra-ui/react/dist/cjs/index.cjs
(web.browser.legacy)
  "@ark-ui/react/steps" in                    
.../node_modules/@chakra-ui/react/dist/cjs/components/steps/steps.cjs
(web.browser.legacy)
  "@ark-ui/react/tabs" in                     
.../node_modules/@chakra-ui/react/dist/cjs/components/tabs/tabs.cjs
(web.browser.legacy)
  "@ark-ui/react/toast" in                    
.../node_modules/@chakra-ui/react/dist/cjs/components/toast/toast.cjs
(web.browser.legacy)
  "@ark-ui/react/toggle" in                   
.../node_modules/@chakra-ui/react/dist/cjs/components/toggle/toggle.cjs
(web.browser.legacy)
  "http2" in .../node_modules/@smithy/node-http-handler/dist-cjs/index.js
(web.browser.legacy)
  "unist-util-visit-parents/do-not-use-color" in
.../node_modules/mdast-util-find-and-replace/node_modules/unist-util-visit-parents/lib/index.js
(web.browser.legacy)
```

**Comment by segunadebayo** [comment]
Please attach a minimal reproduction link

**Reply by danyhiol** [reply]
[reproduction](https://github.com/danyhiol/chakra-text)

**Reply by segunadebayo** [reply]
Looking into it.

**Reply by segunadebayo** [reply]
It seems Meteor doesn't support ESM yet, making it challenging for Chakra to work correctly.
https://github.com/meteor/meteor/issues/11631

I see it's on their roadmap

**Reply by eminx** [reply]
Same issue here.
Meteor has long been supporting ESM packages. I'm pretty sure the case isn't about it...

I had difficulty with this command working in my setup: `npx @chakra-ui/cli snippet add` 
It gives the error: `Error [ERR_MODULE_NOT_FOUND]: Cannot find package '@chakra-ui/react'...`

Could there be a conflict in the way of working of these? I also don't understand where/how the new `Provider` is generated/used replacing the good old `ChakraProvider`... 

Meteor is a monorepo for client and server. Could the issue relate to that? 

Thanks!

**Reply by danyhiol** [reply]
> It seems Meteor doesn't support ESM yet, making it challenging for Chakra to work correctly. [meteor/meteor#11631](https://github.com/meteor/meteor/issues/11631)
> 
> I see it's on their roadmap

this [comment](https://github.com/meteor/meteor/issues/11631#issuecomment-950954230) clearly state the opposite. Not sure if there is any workaround here.

**Reply by danyhiol** [reply]
This issue is also present in `@chakra-ui/react@2.10` with `meteor`. I've not checked the source of the problem. But something definitely changed from some versions.

**Reply by danyhiol** [reply]
I can now confirm that Meteor@3 does not support ESM yet from [this](https://forums.meteor.com/t/upgrade-to-3-complaining-about-legacy-browser-modules/62741) comment and [this](https://github.com/meteor/meteor/pull/13520) pull-request.

**Reply by r1ctn** [reply]
Same problem here ! I had a bunch of : 
Unable to resolve some modules:

  "@ark-ui/react/download-trigger" in .../node_modules/@chakra-ui/react/dist/esm/components/download-trigger/index.js
(web.browser)
  "@ark-ui/react/format" in .../node_modules/@chakra-ui/react/dist/components/format/index.js (web.browser)
...
I fixed it locally by going in thoses files and change the imports. for example : 
from  "@ark-ui/react/download-trigger" to  "@ark-ui/react/dist/components/download-trigger" to follow the ark package organisation.

It feels dirty but everything worked perfectly after

**Comment by segunadebayo** [comment, marked as answer]
This is a Meteor resolver issue. We need a minimal repro. None came.
